sdk/core/azure-core/src/main/java/com/azure/core/util/BinaryData.java

```diff
@@ -224,7 +233,8 @@ public static BinaryData fromObject(Object data) {
      * <p>
      * If {@code data} is null an empty {@link BinaryData} will be returned.
      * <p>
-     * <b>Note:</b> A {@link JsonSerializer} implementation must be available on the classpath.
+     * <b>Note:</b> This method first looks for a {@link JsonSerializer} implementation on the classpath. If no
+     * implementation is found, a default Jackson-based implementation will be used.
```

Comment:
Should we directly mention that it will be based on `JacksonAdapter`?

Reply:
Don't forget to remove the `throws` Javadoc as that no longer can occur.

Reply:
Updated javadoc. I didn't mention JacksonAdapter explicitly to have some flexibility if we want to change the implementation.